ios/RNCallKeep/RNCallKeep.m

```diff
@@ -209,13 +213,53 @@ - (void)dealloc
     _isStartCallActionEventListenerAdded = YES;
 }
 
+RCT_EXPORT_METHOD(reportConnectingOutgoingCallWithUUID:(NSString *)uuidString)
+{
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    [self.callKeepProvider reportOutgoingCallWithUUID:uuid startedConnectingAtDate:[NSDate date]];
+}
+
 RCT_EXPORT_METHOD(reportConnectedOutgoingCallWithUUID:(NSString *)uuidString)
 {
     NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
     [self.callKeepProvider reportOutgoingCallWithUUID:uuid connectedAtDate:[NSDate date]];
 }
 
-RCT_EXPORT_METHOD(setMutedCall:(NSString *)uuidString muted:(BOOL)muted)
+RCT_EXPORT_METHOD(reportEndCallWithUUID:(NSString *)uuidString :(int)reason)
+{
+#ifdef DEBUG
+    NSLog(@"[RNCallKeep][reportEndCallWithUUID] uuidString = %@ reason = %d", uuidString, reason);
+#endif
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    switch (reason) {
+        case CXCallEndedReasonFailed:
+            [self.callKeepProvider reportCallWithUUID:uuid endedAtDate:[NSDate date] reason:CXCallEndedReasonFailed];
+            break;
+        case CXCallEndedReasonRemoteEnded:
+            [self.callKeepProvider reportCallWithUUID:uuid endedAtDate:[NSDate date] reason:CXCallEndedReasonRemoteEnded];
+            break;
+        case CXCallEndedReasonUnanswered:
+            [self.callKeepProvider reportCallWithUUID:uuid endedAtDate:[NSDate date] reason:CXCallEndedReasonUnanswered];
+            break;
+        default:
+            break;
+    }
+}
+
+RCT_EXPORT_METHOD(updateDisplay:(NSString *)uuidString :(NSString *)displayName :(NSString *)uri)
```

Comment:
This method looks like a good candidate to merge with 
`RCT_EXPORT_METHOD(reportUpdatedCall:(NSString *)uuidString...)` on [line 503 ](https://github.com/react-native-webrtc/react-native-callkeep/blob/7b9c1cc6bb461b0787d7a7bac5e7e362bc6ceb66/ios/RNCallKeep/RNCallKeep.m#L503)in this file, we can keep the api back-compat by adding the extra arg.

Additionaly, to make this api would be more flexible if we change `uri => handle` and allow the caller/client application to also pass `handleType`, as the owner of the call session. When we `displayIncomingCall` we specify handleType, which may not always be `initWithType:CXHandleTypePhoneNumber`.

```
-displayIncomingCall
  CXCallUpdate *callUpdate = [[CXCallUpdate alloc] init];
    callUpdate.remoteHandle = [[CXHandle alloc] initWithType:_handleType value:handle];
```